rules/TypeDeclaration/Rector/ClassMethod/ReturnNeverTypeRector.php

```diff
@@ -100,7 +97,12 @@ public function refactor(Node $node): ?Node
             return null;
         }
 
-        $node->returnType = new Name('never');
+        if ($node->returnType && $this->isName($node->returnType, 'never')) {
```

Comment:
we need a explicit 'do-nothing-when-native-never-already-exists' case here, because of the already existing test expecting this behaviour.

before this PR this feature was implicitly supported since rector did not touch the return type which was the same as before with `$node->returnType = new Name('never');`